### PR TITLE
Ensy 172 add functionality and tests to lambda

### PR DESCRIPTION
**Pull request by adamshire123**
### What does this PR do?

Adds functionality and associated tests for transforming the bursar fine and fee data exported from Alma into the format required for uploading to SFS. 

* When invoked with an event containing the job name and job id matching a file in the `SOURCE_BUCKET`, with the correct `SOURCE_PREFIX`, retrieve the file. 
* Convert the file to a .csv according to business rules specified by IDLA and SFS (student financial services)
* Put the transformed file in an s3 bucket where it can be picked up by IDLA staff and manually uploaded to SFS via webform.
* Return the key of the transformed file (we might return something different once we determine how MIT libraries staff will access the transformed file). 

### Helpful background context

This lambda will be invoked by a step function following our pattern of Alma job-end webhook -> webhook lambda -> step function -> this lambda 

The data in the webhook POST request from Alma doesn't contain enough information for our lambda to retrieve the correct object from s3 directly. Alma names the export file [bursar transfer job name]-[job id]-[timestamp].xml. The POST data doesn't contain the timestamp, so we have to first get a list of filenames in our source location, filtered based on the job_name and  job_id from the webhook POST data. There should only be one matching filename because the job_id is unique.

We are still waiting for confirmation on how to calculate some of the values in the transformed file. see https://mitlibraries.atlassian.net/browse/ENSY-182

We are currently putting the transformed file in the sftp bucket under a different path, this may change depending on the outcome of https://mitlibraries.atlassian.net/browse/ENSY-181

### How can a reviewer manually see the effects of these changes?

see readme for local testing instructions

### Includes new or updated dependencies?

YES

### What are the relevant tickets?

https://mitlibraries.atlassian.net/browse/ENSY-172

### Developer

- [x] All new ENV is documented in README (or there is none)
- [ ] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,20 +54,22 @@ bursar's system.
   docker build -t bursar_transfer:latest .
   ```
 
-- Run the default handler for the container:
+- Run the default handler for the container, with required environment variables in `.env`. Make sure the buckets in your `.env` actually exist:
```
Comment (ehanson8) [inline README.md:57]: I would add that the AWS credentials need to be in the `.env` file as well for the container to access AWS

**Review by ehanson8 — CHANGES_REQUESTED** [review]
Works as expected, just a few suggestions

File: README.md [hunk 2]
```diff
@@ -26,20 +54,22 @@ bursar's system.
...
+  docker run --env-file .env -p 9000:8080 bursar_transfer:latest
   ```
 
+- Upload a source file to s3 bucket and folder specified in your `.env`. You can use the file `tests/fixtures/test.xml` rename it to `test_name-test_id-12345.xml`
+
 - Post to the container:
 
   ```bash
-  curl -XPOST "http://localhost:9000/2015-03-31/functions/function/invocations" -d '{}'
+  url -XPOST "http://localhost:9000/2015-03-31/functions/function/invocations" -d '{"job_name":"test_name","job_id":"test_id"}'
```
[the remark this answers is not in the document]
Comment (adamshire123) [reply README.md:68]: isn't `curl` the command? That's how it is in the template https://github.com/MITLibraries/python-lambda-template/blob/6a61d6b134b785d2e9c372415d7d1b7cabd319af/README.md?plain=1#L58

File: lambdas/bursar_transfer.py [hunk 3]
```diff
@@ -23,10 +30,169 @@
...
+            )
+            csv_line["AMOUNT"] = fine_fee.findtext(
+                "xb:compositeSum/xb:sum", default=None, namespaces=name_space
+            )
+            csv_line["EFFECTIVEDATE"] = "effective_date"  # Not sure how to calculate
+            csv_line["BILLINGTERM"] = billing_term(today)
+            if all(csv_line.values()):
+                writer.writerow(csv_line)
+            else:
+                raise ValueError(
```
Comment (ehanson8) [inline lambdas/bursar_transfer.py:147]: Non-blocking but if this scenario happens with any frequency, you might update the error message to include which field is missing for easier troubleshooting.

**Review by ehanson8 — APPROVED** [review]
One non-blocking comment that I defer to you on whether it is useful or not

File: lambdas/bursar_transfer.py (hunk 3 above)
Comment (adamshire123) [reply lambdas/bursar_transfer.py:147]: Yes, I was thinking about that. I don't think this should happen ever, but it is a requirement of SFS that no fields are blank so I thought we should check for it.

**Pull request merged**